### PR TITLE
Added batch usage implementation

### PR DESCRIPTION
**Pull request by ankisho**
Added implementation for subscription emit batch usage and UT

### CODE REVIEW

File: src/SaaS.SDK.Client/Services/MeteredBillingAPIClient.cs [hunk 1]
```diff
@@ -70,5 +71,30 @@ public async Task<MeteringUsageResult> EmitUsageEventAsync(MeteringUsageRequest
...
+        /// <summary>
+        /// Manage Subscription Batch Usage.
+        /// </summary>
+        /// <param name="subscriptionBatchUsageRequest">The subscription batch usage request.</param>
+        /// <returns>
+        /// Subscription Usage
+        /// </returns>
+        public async Task<MeteringBatchUsageResult> EmitBatchUsageEventAsync(IEnumerable<MeteringUsageRequest> subscriptionBatchUsageRequest)
+        {
+            this.Logger?.Info($"Inside ManageSubscriptionUsageAsync() of FulfillmentApiClient, trying to Manage Subscription Batch Usage :: {subscriptionBatchUsageRequest.FirstOrDefault()?.ResourceId}");
```
Comment (dev-kiranbanda) [inline src/SaaS.SDK.Client/Services/MeteredBillingAPIClient.cs:84]: May be we should log the number of items in the request :)

File: src/SaaS.SDK.UnitTest/MeteredApiTest.cs [hunk 2]
```diff
@@ -70,10 +71,43 @@ public async Task TestSubscriptionUsage()
                 Quantity = 5,
                 ResourceId = defaultSubscription.Id
             };
-            var response = this.client.EmitUsageEventAsync(subscriptionUsageRequest).Result;            
+            var response = this.client.EmitUsageEventAsync(subscriptionUsageRequest).Result;
```
Comment (dev-kiranbanda) [inline src/SaaS.SDK.UnitTest/MeteredApiTest.cs:74]:
```csharp

this.client.EmitUsageEventAsync(subscriptionUsageRequest).ConfigureAwait(false).GetAwaiter().GetResult();

```

Comment (ankisho) [reply src/SaaS.SDK.UnitTest/MeteredApiTest.cs:74]: added await and removed the .Result.

File: src/SaaS.SDK.Client/Services/MeteredBillingAPIClient.cs (hunk 1 above)
Comment (ankisho) [reply src/SaaS.SDK.Client/Services/MeteredBillingAPIClient.cs]: Added log for number of request items.

**Pull request merged**